### ISSUE TITLE
clearing/checking extendEvents

### ISSUE THREAD

**Issue by brandonmp** [keen/keen-tracking.js#61]
Is it possible to clear the data set with a call to `extendEvents()`? 

And, related: is it possible to check & see what the current `extendEvents()` inclusions are? eg

```
client.extendEvents(() => {
  return {
    time: Keen.helpers.getDateTimeIndex(),
    userId: a9sjkll3,
    sessionID: Keen.helpers.getUniqueId()
  }
})

// and then:
client.getEventExtensions()
/* returns: 
{
         time: 12345134134,
         userId: a9sjkll3,
         sessionID: <random uuid>
}
*/
```

**Comment by dustinlarimer** [comment]
Not currently possible, but an interesting question/suggestion! Can you share a bit more about what you would like to achieve here?

**Comment by brandonmp** [comment]
sure. i'm running the `keen-tracking` lib in a chrome extension. the extension only affects a few sites via content script injections, so i only care about tracking events on those pages (let's say, `www.example.com`). the flow is something like this:

1. the extension loads and creates a new client, ie `new Keen({})

2. when user lands on an `example.com` page, keen tracks a `session_start` event and calls `extendEvents` w/ some data, eg, `sessionId`, `Keen.utils.timer()`, etc)

3. keen tracks user's interactions w/ `example.com`

4. if 15 minutes pass w/o a pageload of `example.com`, keen fires a `session_end` event

I know from the repo's `readme` that `extendEvent()` calls take precedent over identically named fields from prior calls, but I guess my concern is whether or not the calls just continue to stack-up. 

in other words, if my user keeps browser open for a week and has 100 sessions, is my keen `client` evaluating 100 separate `extendEvent()`s with each `recordEvent()`?

**Comment by dustinlarimer** [comment]
Would this state be reset on every load, or does it persist from page to
page?

**Comment by brandonmp** [comment]
it'd be in the `background script` of the extension, so it'd persist basically until the Chrome process or the extension's process were killed. 

the `content script` that the extension injects into target pages has state that's reset every load, but i'm not executing any tracking code from the `content script` (only via `messages` sent from `content` to `background`

**Comment by dustinlarimer** [comment]
Ok, in this case I would only use `extendEvents` for common props you want
included in every case, and keep all session-specific props in the
`recordEvent` calls. How does that sound?

**Comment by dustinlarimer** [comment]
Ok, in this case I would only use `extendEvents` for common props you want
included in every case, and keep all session-specific props in the
`recordEvent` calls. How does that sound?

**Comment by brandonmp** [comment]
sure that's manageable. can i infer from that guidance that layering `extendEvent` calls indefinitely is an antipattern?

**Comment by dustinlarimer** [comment]
Yeah, definitely causes problems in this case.. the real value of that
method is to remove redundancy and weight from recordEvent calls, with
regard to data modeling.

Glad that approach will work for ya!
